src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java

```diff
@@ -353,15 +353,26 @@ public Response saveAttachment(InputStream in, String name, String contentType,
      * @param updateHandlerUri The Update Handler URI, in the format: <code>designDoc/update1</code>
      * @param docId            The document id to update.
      * @param params           The query parameters as {@link Params}.
+     * @param postBodyRequest  The POST body to create a document if needed.
      * @return The output of the request.
      */
-    public String invokeUpdateHandler(String updateHandlerUri, String docId, Params params) {
+    public String invokeUpdateHandler(String updateHandlerUri, String docId, Params params,
+                                      String postBodyRequest) {
         assertNotEmpty(updateHandlerUri, "uri");
-        assertNotEmpty(docId, "docId");
         final String[] v = updateHandlerUri.split("/");
-        final URI uri = new DatabaseURIHelper(dbUri).path("_design").path(v[0]).path("_update")
-            .path(v[1]).path(docId).query(params).build();
-        final InputStream response = couchDbClient.put(uri);
+        final InputStream response;
+        final URI uri;
+        if(docId != null && !docId.isEmpty()) {
+            uri = new DatabaseURIHelper(dbUri).path("_design").path(v[0]).path("_update")
```

Comment:
A lot of this URI is common between both requests, maybe better to share between the branches? Maybe something like this, but with the comments
```java
DatabaseURIHelper uriHelper = new DatabaseURIHelper(dbUri).path("_design").path(v[0]).path("_update").path(v[1]).query(params);
if (docId != null && docId.isEmpty()) {
  uri = uriHelper.path(docId).build();
  response = couchDbClient.put(uri);
} else {
  uri = uriHelper.build();
  response = couchDbClient.post(uri, postBodyRequest);
}
```